workers/common/src/main/kotlin/common/OrtBaseResultGenerator.kt

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2024 The ORT Server Authors (See <https://github.com/eclipse-apoapsis/ort-server/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.eclipse.apoapsis.ortserver.workers.common
+
+import org.eclipse.apoapsis.ortserver.model.OrtRun
+
+import org.ossreviewtoolkit.model.OrtResult
+
+/**
+ * An internally used helper class to generate an [OrtResult] from the current [OrtRun].
+ *
+ * This class loads the results of the previous worker steps and generates an [OrtResult] from them. In addition, it
+ * adds a number of common labels that can be evaluated by different types of workers to obtain further information
+ * about the run.
+ *
+ * The following labels are added:
+ * - runId: The id of the [OrtRun].
+ */
+object OrtBaseResultGenerator {
```

Comment:
I like the general direction of this refactoring. Just one thought: Would it make sense - instead of introducing a new object - to add the `generate()` function directly to `OrtRunService`? The current implementation suffers from [feature envy](https://refactoring.guru/smells/feature-envy).

Reply:
Yes, I had a look and it makes a lot of sense, as the OrtRunService was "injected" as a parameter and did a lot of things using the OrtRunService. So it is better to have all the code at the same place. I got rid of this OrtBaseResultGenerator and put it into OrtRunService.
Some of the test cases also had then to be moved to OrtRunServiceTest.